Library/Formula/open-mesh.rb

```diff
@@ -16,11 +16,17 @@ class OpenMesh < Formula
   head "http://openmesh.org/svnrepo/OpenMesh/trunk/", :using => :svn
 
   depends_on "cmake" => :build
-  depends_on "qt"
+  depends_on "qt" => :optional
```

Comment:
> To be concrete, I do not know when and where the patch command will be called. Is patch applied on untarred files? under the root dir?

The tarball is untarred into the `buildpath`, and then any patching/inreplaces are applied in the `buildpath` before anything else begins.

> So what's the recommended usage of :optional

Dependencies should always (Ideally) be mandatory, which is just:

```ruby
depends_on "openssl"
```

Recommended (Built with by default, required for bottled installations, available for the user to manually reject) -

```ruby
depends_on "openssl" => :recommended
```

Option dependencies aren't built with by default, not required for either bottled or from-source installs, but available to the user to manually ask `brew` to install with - 

```ruby
depends_on "openssl" => :optional
```

The only real reason dependencies should be `if build.with?` is if they only apply in certain conditions, aren't triggered by the dependency, and asking the user to specify all the deps is meh:

```
depends_on "openssl" => :optional
depends_on "jpeg" if build.with? "openssl"
```

In this case `jpeg` isn't a dependency of `openssl` so it wouldn't be called, but if for some reason the formula requires you to have `jpeg` installed and available to build with `openssl` support the extra dependency needs to be declared as above.

`if build.with?` and `if build.without?` are largely intended for use inside the `def install` block, to declare which arguments should be applied in which situations.

```ruby
if build.with? "openssl"
  args << "--with-openssl=yup"
else
  args << "--without-openssl"
end
```